internal/health_checks/health_check_error.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package health_checks
+
+type HealthCheckError struct {
+	Code         string
+	Class        string
+	Message      string
+	Action       string
+	ResourceLink string
+	IsFatal      bool
+	Err          error
+}
+
+func (e HealthCheckError) Error() string {
+	return e.Err.Error()
+}
+
+var (
+	FB_METRICS_PORT_ERR = HealthCheckError{
+		Code:         "FB_METRICS_PORT_ERR",
+		Class:        "PORT",
+		Message:      "Port 20202 used for fluent-bit self metrics is unavailable.",
+		Action:       "Verify the 20202 port is available.",
+		ResourceLink: "https://cloud.google.com/logging/docs/agent/ops-agent/troubleshooting",
+		IsFatal:      true,
+	}
+	OTEL_METRICS_PORT_ERR = HealthCheckError{
+		Code:         "OTEL_METRICS_PORT_ERR",
+		Class:        "PORT",
+		Message:      "Port 20201 used for open telemetry self metrics is unavailable.",
```

Comment:
Addressed in https://github.com/GoogleCloudPlatform/ops-agent/pull/1051/commits/d2845edbb605e75924e9a00dacde743bd776558b .

Reply:
I thought about this (and the fluent-bit message) some more and I wonder now if it should just say "Port 20201 needed for the Ops Agent is unavailable." We generally don't expect the user to be aware that the Ops Agent runs sub-agents or what their names are, so referring to it as OpenTelemetry might end up being confusing. WDYT? CC @mtabasko

Reply:
Changed both `fluent-bit` and `opentelemetry` names to `Ops Agent` in https://github.com/GoogleCloudPlatform/ops-agent/pull/1051/commits/025b80e1479b7d13c4a33ce1657103e9bbba11f6 . WDYT ? I think it would be less confusing.

Reply:
LGTM 👍